File: ch16/phpunit/dev/work/tests/TaxTest.php
<?php
namespace Root\Root;
use PHPUnit\Framework\TestCase;

final class TaxTest extends TestCase
{
    /* 構文：assertEqual(期待値, 実際の値); */
    public function test_非課税商品は0パーセントの税込金額を返す(): void
    {
        $tax = new Tax;
        $this->assertEquals(1000, $tax->calc_tax(1000, True, False, False));
    }

    public function test_軽減税理対象商品を持ち帰った時は8パーセントの税込金額を返す(): void
    {
        $tax = new Tax;
        $this->assertEquals(1080, $tax->calc_tax(1000, False, True, True));
    }   

    public function test_軽減税理対象商品を店内で飲食した時は10パーセントの税込金額を返す(): void
    {
        $tax = new Tax;
        $this->assertEquals(1100, $tax->calc_tax(1000, False, True, False));
    }   

    public function test_その他の場合は10パーセントの税込金額を返す(): void
    {
        $tax = new Tax;
        $this->assertEquals(1100, $tax->calc_tax(1000, False, False, False));
    }
}

File: ch16/junit/dev/work/app/src/test/java/root/TaxTest.java
package root;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

public class TaxTest {
    @Test
    void 非課税商品は0パーセントの税込金額を返す() {
        Tax tax = new Tax();
        assertEquals(1000, tax.calcTax(1000, true, false, false));
    }

    @Test
    void 軽減税率対象商品を持ち帰った時は8パーセントの税込金額を返す() {
        Tax tax = new Tax();
        assertEquals(1080, tax.calcTax(1000, false, true, true));
    }

    @Test
    void 軽減税率対象商品を店内で飲食した時は10パーセントの税込金額を返す() {
        Tax tax = new Tax();
        assertEquals(1100, tax.calcTax(1000, false, true, false));
    }

    @Test
    void その他の場合は10パーセントの税込金額を返す() {
        Tax tax = new Tax();
        assertEquals(1100, tax.calcTax(1000, false, false, false));
    }
}
